### PR TITLE
Support and facilitate converting text to paths

### PR DESCRIPTION
**Pull request by Timer00**
> A **path text** is text that was converted to paths using a tool such as inkscape. 

### Problem

Rendering text within an SVG may cause issues since text can be rendered differently in different contexts. Simply transforming text to paths resulted in malformed letters:

Before:
image

### Solution

After:
image

Render text as regular paths within our SVGs.

### PR Objectives

- [x] Consistent and readable rendering of path texts
- [x] Little to no changes in current symbol creation flow for Symbols that include text

### Converting text to paths
[Short video on How to convert Texts to Paths](https://www.loom.com/share/cb1b9ad39a444676a90feb3e24b6e48b?sid=4e906d1c-740a-41bc-9ab1-1539040724c4)

### Set labels for selected paths
image

Select multiple paths in your inkscape layers -> Open Extensions -> Custom -> Set labels for selected paths.
image
In the dialog that shows up you can set your label and optional numbering which will apply to all selected paths.

### CODE REVIEW

**Comment by Timer00** [comment]
### Current status
After considerable tweaking I found a way to render path texts in a readable and somewhat consistent representation. However, this rendering requires modifying the strokeWidth, fill and closed properties of only the text paths.
So currently I am looking into ways we could detect that a path coming from inkscape was originally text, and then apply the correct styling.

**Comment by Timer00** [comment]
> ### Current status
> After considerable tweaking between code and inkscape I found a way to render path texts in a readable and somewhat consistent representation. However, this rendering requires modifying the strokeWidth, fill and closed properties of only the text paths. So currently I am looking into ways we could detect that a path coming from inkscape was originally text, and then apply the correct styling.
> 
> If you have any other ideas feel free to let me know!

I've solved this challenge by integrating the inkscape:label into our Primitives and implementing an extension for batch labeling + another for flattening groups recursively [wip] in inkscape.

**Comment by Timer00** [comment]
Will remove text sample once approved.

File: generated/symbols-index.ts [hunk 1]
```diff
@@ -86,6 +88,8 @@ export default {
   "schottky_diode_vert": _schottky_diode_vert,
   "silicon_controlled_rectifier_horz": _silicon_controlled_rectifier_horz,
   "silicon_controlled_rectifier_vert": _silicon_controlled_rectifier_vert,
+  "text_paths_sample_horz": _text_paths_sample_horz,
+  "text_paths_sample_vert": _text_paths_sample_vert,
```
Comment (seveibar) [inline generated/symbols-index.ts:92]: we can use a snapshot test outside of the regular symbols definitions using https://github.com/tscircuit/bun-match-svg, that way this doesn't increase bundle size for symbols or get mixed in with electrical symbols

File: scripts/inkscape/organize_svg.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                # Move the element without modifying its transform
+                element.append(desc)
+        
+        # Remove all groups
+        for desc in list(descendants):  # Create a new list to avoid modifying while iterating
+            if desc.tag == '{http://www.w3.org/2000/svg}g' and desc != element:
+                desc.getparent().remove(desc)
+
+if __name__ == '__main__':
+    OrganizeSVG().run()
```
Comment (seveibar) [inline scripts/inkscape/organize_svg.py:70]: i appreciate the effort on this one, but doesn't it make more sense to convert svg text to paths automatically if they aren't {REF} or {VAL} in typescript? The maintainability of python in a ts repo is low

**Review by seveibar — CHANGES_REQUESTED** [review]
absolutely awesome feature but i'd like to remove inkscape as a dependency rather than bake it in so closely, also having python in this repo is likely to decrease maintainability (although ai generally helps to keep maintainability high)

I think we could just as easily do this in Typescript for non-reference text. We already have a custom alphabet defined for PCBs here: https://github.com/tscircuit/pcb-viewer/blob/main/src/assets/alphabet.ts which we could isolate into a dedicated module if need be

**Comment by seveibar** [comment]
RE: alphabet, if that is useful we can literally just copy it over for now.

**Comment by seveibar** [comment]
just watched the video. It's kind of interesting because I can definitely see us having inkscape extensions to improve inkscape workflows. It makes a lot of sense, but it also kind of creates this extra knowledge layer/tooling dependency vs doing it programmatically for non-reference text.

But a quick comment on the video: ABSOLUTELY AWESOME. Tutorial videos are a huge value add

**Comment by Timer00** [comment]
> absolutely awesome feature but i'd like to remove inkscape as a dependency rather than bake it in so closely, also having python in this repo is likely to decrease maintainability (although ai generally helps to keep maintainability high)
> 
> I think we could just as easily do this in Typescript for non-reference text. We already have a custom alphabet defined for PCBs here: https://github.com/tscircuit/pcb-viewer/blob/main/src/assets/alphabet.ts which we could isolate into a dedicated module if need be

oh man! I had this same idea but thought it would be too much work to create a whole alphabet. That definitely makes it easier and this solutions is way less hassle for our fellow contributors. Will def work on that soon!

**Comment by Timer00** [comment]
One thought that occurred to me is that this could still be useful for converting any UTF-8 character to a path.

**Comment by seveibar** [comment]
yea i think we could do the conversion in this module, or import it as an npm module. We should have some snapshot testing on it to make sure it's good.

(I think it might be better to implement inside this module vs externally to make things easier for contributors if there's a bug)

**Pull request closed without merging**